packages/store/src/internal/state-operations.ts

```diff
@@ -38,6 +43,54 @@ export class InternalStateOperations {
     return rootStateOperations;
   }
 
+  public ngxsBootstrap(options: NgxsInitializeOptions) {
+    const { factory, states, action, ngxsAfterBootstrap } = options;
+    const results: DefaultStateRef = factory.addAndReturnDefaults(states || []);
+    const stateRootOperations = this.getRootStateOperations();
+
+    const currentStateByRootTree = stateRootOperations.getState();
+    const newStateBySubTree = results.defaults;
+
+    const rootStatesKeys: string[] = Object.keys(currentStateByRootTree);
+    const newStatesKeys: string[] = Object.keys(newStateBySubTree);
+
+    const unmountedKeys: string[] = newStatesKeys.filter(
+      name => !rootStatesKeys.includes(name)
+    );
+
+    const uniqueResult: DefaultStateRef = this.findUnmountedState(results, unmountedKeys);
+    const canBeUpdateStateTree: boolean = uniqueResult.states.length > 0;
+
+    if (ngxsAfterBootstrap) {
+      ngxsAfterBootstrap();
+    }
+
+    if (canBeUpdateStateTree) {
+      stateRootOperations.setState({ ...currentStateByRootTree, ...uniqueResult.defaults });
+      stateRootOperations.dispatch(new action()).subscribe(() => {
+        factory.invokeInit(uniqueResult.states);
+      });
+    }
+  }
+
+  private findUnmountedState(result: DefaultStateRef, uniqueKeys: string[]): DefaultStateRef {
+    const newResult: DefaultStateRef = { defaults: {}, states: [] };
+    const defaults = result.defaults;
+    const states: MappedStore[] = result.states;
+
+    for (const options in defaults) {
+      if (defaults.hasOwnProperty(options) && uniqueKeys.includes(options)) {
+        newResult.defaults[options] = defaults[options];
+      }
+    }
+
+    newResult.states = states.filter((meta: MappedStore) => {
+      return uniqueKeys.some((key: string) => meta.depth.includes(key));
```

Comment:
```suggestion
      return uniqueKeys.some((key: string) => meta.depth.split('.')[0] === key);
```
This is better because it checks only the first level to match the key, as opposed to any level.